hypothesis-python/tests/cover/test_direct_strategies.py

```diff
@@ -33,6 +33,21 @@
 from tests.common.utils import checks_deprecated_behaviour
 
 
+def stable_repr(x):
+    """Wrapper around ``nicerepr`` that also forces sets and dicts to
+    print in a consistent order across multiple processes, even if they have
+    different values for PYTHONHASHSEED.
+    """
+    if isinstance(x, dict):
+        items = [("%s=%s" % (k, stable_repr(v))) for k, v in x.items()]
+        return "{%s}" % (", ".join(sorted(items)),)
+    elif isinstance(x, set) and len(x) > 0:
+        values = [stable_repr(v) for v in x]
+        return "{%s}" % (", ".join(sorted(values)),)
+    else:
+        return nicerepr(x)
```

Comment:
This isn't foolproof, since `x` could be some other collection type that contains a set/dict.

But since this is test-only code, I think we can worry about that if and when it actually happens.

Reply:
We could switch this over to using the pretty printing from `hypothesis.vendor.pretty`, which I think already sorts sets and dict keys?

Reply:
That seems a lot easier than doing this by hand!